src/app/features/worklog/worklog-export/worklog-export.util.ts

```diff
@@ -152,40 +49,130 @@ export const createRows = (tasks: WorklogTask[], startTimes: WorkStartEnd, endTi
     // TODO check all the typing
     group.titles = unique<any>(
       group.tasks.map((t: WorklogTask) => (
-        t.parentId && (tasks.find(ptIN => ptIN.id === t.parentId) as WorklogTask).title
+        t.parentId && (data.tasks.find(ptIN => ptIN.id === t.parentId) as WorklogTask).title
       ) || (!t.parentId && t.title)
       )
     ).filter((title: string) => !!title);
-    group.dates = unique(group.dates).sort((a: string, b: string) => {
-      const dateA: number = new Date(a).getTime();
-      const dateB: number = new Date(b).getTime();
-      if (dateA === dateB) {
-        return 0;
-      } else if (dateA < dateB) {
-        return -1;
-      }
-      return 1;
-    });
+    group.dates = unique(sortDateStrings(group.dates));
 
     group.notes = group.notes.map((note) => {
       return note.replace(/\n/g, ' - ');
     });
 
     group.projects = unique(group.projects);
     group.projects = group.projects.map((pId: string) => {
-      return (allProjects.find(project => project.id === pId) as ProjectCopy).title;
+      return (data.projects.find(project => project.id === pId) as ProjectCopy).title;
     });
 
     group.tags = unique(group.tags);
     group.tags = group.tags.map((tId: string) => {
-      return (allTags.find(tag => tag.id === tId) as TagCopy).title;
+      return (data.tags.find(tag => tag.id === tId) as TagCopy).title;
     });
 
     rows.push(group);
   }));
 
   return rows;
+};
+
```

Comment:
I would love to see a short introductory comment on what this does and how, because I am pretty sure I will quickly forget this :)